File: OOP_variable/3.py
class Cal(object):
    def __init__(self, v1, v2):
        if isinstance(v1, int):
            self.__v1 = v1
        if isinstance(v2, int):
            self.v2 = v2
    
    def add(self):
        return self.__v1 + self.v2

    def subtract(self):
        return self.__v1 - self.v2


c1 = Cal(30, 20)

print(c1.add())
print(c1.subtract())

print(c1.v2)
#print(c1.__v1)

File: OOP_variable/3.rb
class Cal
    # attr_reader :v1
    # attr_writer :v1
    attr_accessor :v1

    def initialize(v1, v2)
       @v1 = v1
       @v2 = v2
    end

    def add()
        return @v1 + @v2
    end
    def subtract()
        return @v1 - @v2
    end
end

c1 = Cal.new(10, 10)

p(c1.add())
p c1.subtract()
c1.v1 = 50
p c1.v1



